server/planning/content_profiles/planning_types_async_service.py

```diff
@@ -0,0 +1,132 @@
+from typing import Any
+from copy import deepcopy
+
+from superdesk.core.resources import AsyncResourceService
+from superdesk.core.types import SearchRequest, ProjectedFieldArg
+from superdesk.utils import ListCursor
+
+from planning.common import planning_link_updates_to_coverage, get_config_event_related_item_search_provider_name
+from planning.types import PlanningTypesResourceModel
+from .profiles import DEFAULT_PROFILES
+
+
+class PlanningTypesAsyncService(AsyncResourceService[PlanningTypesResourceModel]):
+    """Planning types async service
+
+    Provide a service that returns what fields should be shown in the edit forms in planning, in the edit dictionary.
+    Also provide a schema to allow the client to validate the values entered in the forms.
+    Entries can be overridden by providing alternates in the planning_types mongo collection.
+    """
+
+    async def find_one(
+        self,
+        req: SearchRequest | None = None,
+        projection: ProjectedFieldArg | None = None,
+        use_mongo: bool = False,
+        version: int | None = None,
+        **lookup,
+    ) -> PlanningTypesResourceModel | None:
+        """
+        Overrides the `find_one` method to merge default planning type configurations
+        with database entries. If no entry exists in the database, it returns a default
+        planning type configuration.
+        """
+        try:
+            search_request = (
```

Comment:
Would it be possible to pass `req`, `projection`, `use_mongo`, `version` and `lookup` directly to the parent function, as opposed to re-constructing the req object here (as the parent function does this for us already)?
Or does this cause any issues, such as types?

Reply:
I had actually tried that initially but it was causing typing issues because of incompatibility with the base class signature.

Reply:
That's what I thought, oh well